src/main/java/LinkedQueue.java

```diff
@@ -0,0 +1,125 @@
+import java.util.NoSuchElementException;
+
+public class LinkedQueue<T> implements Queue<T> {
+
+    private Node<T> front;
+    private Node<T> rear;
+    private int size;
+
+    public LinkedQueue() {
+        front = null;
+        rear = null;
+        size = 0;
+    }
+
+    @Override
+    public void enqueue(T element) {
+        Node<T> toEnqueue = new Node<>(element);
+
+        // If nothing is in the queue, then the front is null
+        // and the enqueued element will become the front,
+        // otherwise, the enqueued element is added after the
+        // current rear.
+        if (isEmpty()) {
+            front = toEnqueue;
+        }
+        else {
+           rear.setNext(toEnqueue);
+        }
+        // Either way, the rear of the queue will be the newly
+        // enqueued element.
+        rear = toEnqueue;
+        size++;
+    }
+
+    @Override
+    public T dequeue() {
+        if (isEmpty()) {
+            throw new NoSuchElementException("Dequeueing from an empty queue.");
+        }
+        T returnElement = front.getData();
+        front = front.getNext();
+        size--;
+        // Although front will update properly regardless, rear
+        // will need to be set to null if we dequeue the last
+        // element.
+        if (isEmpty()) {
+            rear = null;
+        }
```

Comment:
This isn't wholly true. Don't get me wrong, you _should_ null out the `rear` for garbage collection, but adding an element will do `rear = toEnqueue`, cleaning up the internal state.

Reply:
I added a note about this in #112 mentioning not doing this and messing up a count. 

To include your point however, I will also add a mention of this being good for garbage collection to #112 and change this comment to _should_ from _will need to_.